src/Data/Product.agda

```diff
@@ -113,6 +114,18 @@ map₂ : ∀ {A : Set a} {B : A → Set b} {C : A → Set c} →
        (∀ {x} → B x → C x) → Σ A B → Σ A C
 map₂ f = map id f
 
+-- A version of map where the output can depend on the input
+dmap : ∀ {B : A → Set b} {P : A → Set p} {Q : ∀ {a} → P a → B a → Set q} →
+       (f : (a : A) → B a) → (∀ {a} (p : P a) → Q p (f a)) →
+       (ap : Σ A P) → Σ (B (proj₁ ap)) (Q (proj₂ ap))
+dmap f g (x , y) = f x , g y
+
+-- dmap for non-dependent products
+_<*>_ : ∀ {x y} {X : A → Set x} {Y : B → Set y} →
+        ((a : A) → X a) × ((b : B) → Y b) →
+        (ab : A × B) → X (proj₁ ab) × Y (proj₂ ab)
+(f , g) <*> xy = dmap f g xy
```

Comment:
If we stick with the naming conventions used in this file, shouldn't this be called `dmap′` and should go with the other non-dependant synonyms at the bottom of the file?